### DISCUSSION TITLE
TTL Pulse Generation with Non-buffered Digital Output on NI Devices

### DISCUSSION THREAD

**Discussion by rachelmcassidy** [bonsai-rx/bonsai#2062] [Q&A]
I am using the daqmx package to interface with a NI PXI-6221. I'd like to deliver a precisely timed ~40 ms TTL pulse, but the timing of single-sample buffers coming from the FunctionGenerator node does not seem reliable enough. My device unfortunately does not support buffered operations on DIO lines.

Has there been any progress on implementing counter functionality in the daqmx package (https://github.com/bonsai-rx/daqmx/issues/15)? Is it possible currently with the DigitalOutput node using the SignalSource property if I set up a counter task outside of Bonsai?

Apologies for the duplicate post - I also raised this as an issue in the daqmx repository.

**Comment by bruno-f-cruz** [comment]
You should be able to copy-paste the example I provided in that issue to an [Extension](https://bonsai-rx.org/docs/articles/scripting-extensions.html). Make sure you add the daqmx package reference to the cproj file. We have been looking into ways to improve the daqmx package with more plug and play operators but it would be nice to have people test these local recipes to get a sense of what works and doesn't.

**Reply by rachelmcassidy** [reply]
I'm just getting around to attempting this solution. I'm stuck on one error that I'm hoping is a quick fix:

This line:
`public override IObservable<Unit> Generate()`

generates the following error during the script extension build:
`error CS0246: The type or namespace name 'Unit' could not be found (are you missing a using directive or an assembly reference?)`

Is there another namespace I should reference? Here are my scripts -
[DAQmx_CounterTiming.zip](https://github.com/user-attachments/files/18314336/DAQmx_CounterTiming.zip)

**Reply by bruno-f-cruz** [reply]
Interesting. I cant fully reproduce as I dont have nimax installed on my PC. But If I remove everything from the .cs file but leaving the Unit code it works, so I suppose it is something to do with the namespaces included in your project. Try specifying the Unit type with its namespace explicit: System.Reactive.Unit
